bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/handler/VehicleStateDescriptionProvider.java

```diff
@@ -12,31 +12,57 @@
+
 import org.eclipse.jdt.annotation.NonNullByDefault;
-import org.eclipse.smarthome.core.thing.binding.BaseDynamicStateDescriptionProvider;
-import org.eclipse.smarthome.core.thing.i18n.ChannelTypeI18nLocalizationService;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
 import org.eclipse.smarthome.core.thing.type.DynamicStateDescriptionProvider;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.eclipse.smarthome.core.types.StateOption;
 import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.Deactivate;
 
 /**
  * Dynamic provider of state options for VehicleHandler.
  *
- * @author Gregory Moyer - Initial contribution
- * @author Gaël L'hopital - Copied as-is in VolvoOnCall binding
+ * @author Ga√ęl L'hopital - Initial Contribution (for OH 2.4.0)
  */
 @Component(service = { DynamicStateDescriptionProvider.class, VehicleStateDescriptionProvider.class })
 @NonNullByDefault
-public class VehicleStateDescriptionProvider extends BaseDynamicStateDescriptionProvider {
+public class VehicleStateDescriptionProvider implements DynamicStateDescriptionProvider {
+    private final Map<ChannelUID, List<StateOption>> channelOptionsMap;
+
+    public VehicleStateDescriptionProvider() {
+        this.channelOptionsMap = new ConcurrentHashMap<ChannelUID, List<StateOption>>();
+    }
+
+    public void setStateOptions(final ChannelUID channelUID, final List<StateOption> options) {
+        this.channelOptionsMap.put(channelUID, options);
+    }
 
-    @Reference
-    protected void setChannelTypeI18nLocalizationService(
-            final ChannelTypeI18nLocalizationService channelTypeI18nLocalizationService) {
-        this.channelTypeI18nLocalizationService = channelTypeI18nLocalizationService;
+    @Override
+    public @Nullable StateDescription getStateDescription(final Channel channel,
+            final @Nullable StateDescription original, final @Nullable Locale locale) {
+        final List<StateOption> options = this.channelOptionsMap.get(channel.getUID());
+        if (options == null) {
+            return null;
+        }
+        if (original != null) {
+            return new StateDescription(original.getMinimum(), original.getMaximum(), original.getStep(),
+                    original.getPattern(), original.isReadOnly(), options);
+        }
+        return new StateDescription((BigDecimal) null, (BigDecimal) null, (BigDecimal) null, (String) null, false,
+                options);
     }
 
-    protected void unsetChannelTypeI18nLocalizationService(
-            final ChannelTypeI18nLocalizationService channelTypeI18nLocalizationService) {
-        this.channelTypeI18nLocalizationService = null;
+    @Deactivate
+    public void deactivate() {
+        this.channelOptionsMap.clear();
     }
-}
+}
```

Comment:
You can not import a v2.4 file in a v2.5 build, or I do not get your point.

Reply:
Corrected: New to git / pull requests etc, so I was hoping it was possible to skip some parts included in the pull request on the master side. Have now corrected pull request with the 2.5.0 file.